Library/Homebrew/utils.rb

```diff
@@ -130,8 +130,13 @@ def odeprecated(method, replacement = nil, options = {})
   # - Location outside of 'compat/'.
   # - Location of caller of deprecated method (if all else fails).
   backtrace = options.fetch(:caller, caller)
+  tap_message = nil
   caller_message = backtrace.detect do |line|
-    line.start_with?("#{HOMEBREW_LIBRARY}/Taps/")
+    if line =~ %r{#{Regexp.escape HOMEBREW_LIBRARY}/Taps/([^/]+/[^/]+)/}
```

Comment:
I'm sorry to have to be annoying, but the regular expression is still not anchored at the start (unlike what I suggested), thus is a regression on the `start_with?` used before.